.github/workflows/infra-ci.yml

```diff
@@ -0,0 +1,68 @@
+name: Infrastructure CI
+# the commented out bits should be refined for later so that terraform can be deployed programatically somehow
+
+on:
+  push:
+    paths:
+      - '**.tf'
```

Comment:
you probably want to run it against the `main` branch as well, just in case there are any weird behaviors between different PRs that get merged around the same time.